### DISCUSSION TITLE
Set different kwh cost for day/night

### DISCUSSION THREAD

**Discussion by thunder2k** [teslamate-org/teslamate#1726] [Ideas]
**Describe the bug**

It would be great to be able to set a different kwh cost in relation to the time. I have a tariff which is different between day and night, so every time I charge the car during daytime, I have to manually fix the records with the day tariff. 

**Expected behavior**

**How to reproduce it (as minimally and precisely as possible)**:

**Relevant entries from the logs**

```
paste logs here
```

**Screenshots**

**Data**

**Environment**

- TeslaMate version:
- Type of installation: Docker or Manual
- OS TeslaMate is installed on:
- User OS & Browser:
- Others:

**Comment by tobiasehlert** [comment]
Hi @thunder2k,

You should checkout the **TeslaMateAgile** project, which might be exactly doing what you want to do..
TeslaMateAgile supports that energy providers / tarriffs are either Octopus Agile, Tibber or **fixed pricing** (manually specified).

Link to [TeslaMate docs of projects](https://docs.teslamate.org/docs/projects#teslamateagile)
Link to [TeslaMateAgile on Github](https://github.com/MattJeanes/TeslaMateAgile)

Regards,
Tobias

**Reply by thunder2k** [reply]
> Hi @thunder2k,
> 
> You should checkout the **TeslaMateAgile** project, which might be exactly doing what you want to do..
> TeslaMateAgile supports that energy providers / tarriffs are either Octopus Agile, Tibber or **fixed pricing** (manually specified).
> 
> Link to [TeslaMate docs of projects](https://docs.teslamate.org/docs/projects#teslamateagile)
> Link to [TeslaMateAgile on Github](https://github.com/MattJeanes/TeslaMateAgile)
> 
> Regards,
> Tobias

Thanks for the suggestion, I've installed it but didn't try it yet as I have my car to the body show for Ceramic Coating.

A couple of things are not clear though:

1. How does it deal with daylight saving time?
2. What happens if your charging session spans over 2 different tariffs?

**Reply by tobiasehlert** [reply]
I can't answer to that, but I think @MattJeanes can 😃

**Reply by AmyJeanes** [reply]
Yes my project will do exactly what you need! It will handle daylight savings (assuming you set the correct timezone) and a charge session spanning multiple tariffs no problem at all 🙂

**Reply by thunder2k** [reply]
> Yes my project will do exactly what you need! It will handle daylight savings (assuming you set the correct timezone) and a charge session spanning multiple tariffs no problem at all 🙂

How should I set my tariff interval? Based on NON DST times and then it adjusts automatically?
Example:
during DST cheap tariff is from 00:16am to 9:16am (my meter is off 16 minutes)
when NOT on DST, cheap tariff is from 11:16pm to 8:16pm.
The meter clock never change!
This is how I set my paramenter (I'm in Dublin, Ireland):
- FixedPrice__TimeZone=Europe/London # IANA (tz database) time zone code, used for below times
- FixedPrice__Prices__0=23:16-08:16=0.0523 # Night
- FixedPrice__Prices__1=08:17-23:15=0.1818 # Day

Also, how do you handle the charge across tariffs? Does it split it into 2 different charging processes?

**Reply by AmyJeanes** [reply]
Ah I see, check the tz database to see if there is a timezone you can use which is not affected by DST that could work for you. If not raise an issue on my project and I'll see what we can do

**Reply by AmyJeanes** [reply]
Also the charging process is created by TeslaMate, my project just adds a cost to it and can span multiple times no problem

**Reply by AmyJeanes** [reply]
You could use UTC actually if you're based in the UK as GMT without daylight savings is the same as UTC

**Reply by thunder2k** [reply]
> You could use UTC actually if you're based in the UK as GMT without daylight savings is the same as UTC
Cool, so my config should be:
FixedPrice__TimeZone=UTC # IANA (tz database) time zone code, used for below times
FixedPrice__Prices__0=23:16-08:16=0.0523 # Night
FixedPrice__Prices__1=08:17-23:15=0.1818 # Day

?

**Reply by AmyJeanes** [reply]
The timezone would be Etc/UTC but otherwise yep!

**Reply by thunder2k** [reply]
> The timezone would be Etc/UTC but otherwise yep!

Cool, I've changed it! How do I restart just your plugin and not all teslamate?

Thanks a lot for your help!

**Reply by AmyJeanes** [reply]
Yep just the plugin, if you run into any issues raise an issue on my project 🙂

Glad I could help!

**Reply by thunder2k** [reply]
> Yep just the plugin, if you run into any issues raise an issue on my project 🙂
> 
> Glad I could help!

How do I do it? I usually do
docker-composer up - d but that would restart everything, wouldn't it?

**Reply by AmyJeanes** [reply]
That only restarts apps that have changed configuration so you should be good 👍

The exact command is `docker-compose up -d`

**Comment by adriankumpf** [comment]
This is not a bug. Feature requests belong here: [💡 Ideas](https://github.com/adriankumpf/teslamate/discussions/categories/ideas). I'll move it.

**Comment by enigma2k** [comment]
I agree, this would be a great feature. many people have different energy costs depending on the time

**Reply by tobiasehlert** [reply]
@enigma2k, I don't think it's a feature that should be in TeslaMate directly.. there is a plugin that can solve this for you instead 😃 
https://github.com/adriankumpf/teslamate/discussions/1726#discussioncomment-872444

**Reply by nottheedge** [reply]
There is a plug-in? I’d be interested as I’m charged on hourly base.

**Reply by tobiasehlert** [reply]
@nottheedge with plugin I mean the TeslaMateAgile project :)
